strawberryfields/backends/gaussianbackend/backend.py

```diff
@@ -102,24 +119,42 @@ def measure_homodyne(self, phi, mode, select=None, **kwargs):
         Returns:
             float: measured value
         """
-        eps = kwargs.get('eps', 0.0002)
+        if shots > 1:
+            if select is not None:
+                raise NotImplementedError("Gaussian backend currently does not support "
+                                          "postselection if shots > 1 for homodyne measurement")
+
+            raise NotImplementedError("Gaussian backend currently does not support "
+                                      "shots > 1 for homodyne measurement")
 
         # phi is the rotation of the measurement operator, hence the minus
         self.circuit.phase_shift(-phi, mode)
 
         if select is None:
-            qs = self.circuit.homodyne(mode, eps)[0]
+            qs = self.circuit.homodyne(mode, **kwargs)[0, 0]
         else:
-            val = select * 2/sqrt(2*self.circuit.hbar)
-            qs = self.circuit.post_select_homodyne(mode, val, eps)
+            val = select * 2 / sqrt(2 * self.circuit.hbar)
+            qs = self.circuit.post_select_homodyne(mode, val, **kwargs)
+
+        return qs * sqrt(2 * self.circuit.hbar) / 2
+
+    def measure_heterodyne(self, mode, shots=1, select=None):
 
-        return qs * sqrt(2*self.circuit.hbar)/2
+        if shots > 1:
+            if select is not None:
+                raise NotImplementedError("Gaussian backend currently does not support "
+                                          "postselection if shots > 1 for heterodyne measurement")
+
+            raise NotImplementedError("Gaussian backend currently does not support "
+                                      "shots > 1 for heterodyne measurement")
 
-    def measure_heterodyne(self, mode, select=None):
         if select is None:
             m = identity(2)
-            res = 0.5*self.circuit.measure_dyne(m, [mode])
-            return res[0]+1j*res[1]
+            res = 0.5 * self.circuit.measure_dyne(m, [mode], shots=shots)
+            if shots == 1:
+                return res[0, 0] + 1j * res[0, 1]
```

Comment:
I suppose this will be deleted if we go to all measurements returning an array of shape `(shots, modes)` even if `shots==1`.

Reply:
good catch, that's just an artifact; since `shots !=1` is no supported atm, no need for conditional or other return yet

Reply:
fixed